src/network.c

```diff
@@ -87,12 +54,9 @@ static gboolean transfer(RaucTransfer *xfer, GError **error)
 	//curl_easy_setopt(curl, CURLOPT_MAX_RECV_SPEED_LARGE, 1048576L); /* bytes per second */
 	//curl_easy_setopt(curl,  CURLOPT_LOW_SPEED_LIMIT, 1024L);
 	//curl_easy_setopt(curl,  CURLOPT_LOW_SPEED_TIME, 60L);
-	curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, write_cb);
-	curl_easy_setopt(curl, CURLOPT_WRITEDATA, xfer);
-	curl_easy_setopt(curl, CURLOPT_XFERINFOFUNCTION, xfer_cb);
-	curl_easy_setopt(curl, CURLOPT_XFERINFODATA, xfer);
 	curl_easy_setopt(curl, CURLOPT_FAILONERROR, xfer);
 	curl_easy_setopt(curl, CURLOPT_ERRORBUFFER, errbuf);
+	curl_easy_setopt(curl, CURLOPT_MAXFILESIZE, xfer->limit);
```

Comment:
See https://curl.haxx.se/libcurl/c/CURLOPT_MAXFILESIZE.html:
> The file size is not always known prior to download, and for such files this option has no effect even if the file transfer ends up being larger than this given limit. This concerns both FTP and HTTP transfers.

So we still need to track the actual received file size, but just report errors explicitly, maybe via a flag in xfer.

`CURLOPT_MAXFILESIZE` (or `CURLOPT_MAXFILESIZE_LARGE` since curl 7.11 in 2004) would still be useful for early aborts if the server reports the size.

Reply:
I've pushed an update that does not remove xfer callbacks anymore, but adds basic error message reporting instead.

Reply:
We should use https://curl.haxx.se/libcurl/c/CURLOPT_MAXFILESIZE_LARGE.html here and change the type of xfer->limit accordingly. This would also get rid of some casts in the following patch as well removing the limit on 32bit systems.